app/gui/src/project-view/components/GraphEditor/widgets/WidgetText.vue

```diff
@@ -64,10 +72,17 @@ function makeLiteralFromUserInput(value: string): Ast.Owned<Ast.MutableTextLiter
   }
 }
 
-const shownLiteral = computed(() => inputTextLiteral.value ?? emptyTextLiteral.value)
-const closeToken = computed(() => shownLiteral.value.close ?? shownLiteral.value.open)
+const openToken = computed(() => inputTextLiteral.value?.open ?? emptyTextLiteral.value.open)
+const closeToken = computed(() => inputTextLiteral.value?.close ?? openToken.value)
 
-const textContents = computed(() => shownLiteral.value.rawTextContent)
+const textContents = computed(() =>
+  props.input.value instanceof Ast.TextLiteral ? props.input.value.rawTextContent : '',
+)
+const placeholder = computed(() =>
+  WidgetInput.isPlaceholder(props.input) ?
```

Comment:
No, inputTextLiteral is undefined also in case of `value` not being a text literal (for example, the default value which is a call to a function - we want to also display this call in placeholder).